types/tools/hook-events.d.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Event detail for block relocation
+ */
+export interface MoveEventDetail {
+  /**
+   * index the block was moved from
+   */
+  fromIndex: number;
+  /**
+   * index the block was moved to
+   */
+  toIndex: number;
+}
+
+/**
+ * Move event for block relocation
+ */
+export interface MoveEvent {
+  readonly detail: MoveEventDetail;
```

Comment:
I don't quite understand, why we need this `detail` wrapper?

Reply:
This is common `Event` structure

Reply:
Not simple Event i suppose, but CustomEvent structure:
https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent

Reply:
@Infonautica yes, that is what I'm talking about. 
@natterstefan if you want to create an event-like structure, it's better to use CustomEvent.